### PR TITLE
[Communication] - Phone Numbers - Add browse API methods

### PR DESCRIPTION
**Pull request by danielortega-msft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 1]
```diff
@@ -246,16 +275,151 @@ def get_purchased_phone_number(
...
+        """
+        return self._phone_number_client.phone_numbers.list_offerings(
+            country_code,
+            phone_number_type=kwargs.pop("phone_number_type", None),
+            assignment_type=kwargs.pop("assignment_type", None),
+            **kwargs
+        )
+
+    @distributed_trace
+    def list_available_area_codes(
```
Comment (danielortega-msft) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:385]: Keep in mind that this is the only SDK in which this is not separated into one method for geographic and another one for toll free area codes.

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def delete_trunk(
             **kwargs)
 
     @distributed_trace
-    def get_trunks(
+    def list_trunks(
```
Comment (danielortega-msft) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py:155]: Is this change intended to be part of this PR?

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py [hunk 3]
```diff
@@ -162,10 +162,10 @@ def get_trunks(
         :rtype: Iterable[~azure.communication.siprouting.models.SipTrunk]
         :raises: ~azure.core.exceptions.HttpResponseError
         """
-        return self._get_trunks_(**kwargs)
+        return self._list_trunks_(**kwargs)
 
     @distributed_trace
-    def get_routes(
+    def list_routes(
```
Comment (danielortega-msft) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/siprouting/_sip_routing_client.py:168]: Is this change intended to be part of this PR?

File: sdk/communication/azure-communication-phonenumbers/test/test_phone_number_administration_client.py [hunk 4]
```diff
@@ -274,19 +275,82 @@ def test_update_phone_number_capabilities_with_invalid_number(self, **kwargs):
...
+            "US", PhoneNumberType.TOLL_FREE, PhoneNumberAssignmentType.APPLICATION)
+        assert area_codes.next()
 
-        with pytest.raises(ValueError) as ex:
-            self.phone_number_client.begin_update_phone_number_capabilities(
-                phone_number,
-                PhoneNumberCapabilityType.INBOUND_OUTBOUND,
-                PhoneNumberCapabilityType.INBOUND,
-                polling=True
-            )
```
Comment (danielortega-msft) [inline sdk/communication/azure-communication-phonenumbers/test/test_phone_number_administration_client.py:292]: Why was this removed? This is validating logic that was implemented in a prior PR #28163

File: sdk/communication/azure-communication-phonenumbers/test/test_phone_number_administration_client_async.py [hunk 5]
```diff
@@ -278,36 +280,136 @@ async def test_update_phone_number_capabilities_with_unauthorized_number(self):
...
+        assert len(items) > 0
 
-        with pytest.raises(ValueError) as ex:
-            async with self.phone_number_client:
-                await self.phone_number_client.begin_update_phone_number_capabilities(
-                    phone_number,
-                    PhoneNumberCapabilityType.INBOUND_OUTBOUND,
-                    PhoneNumberCapabilityType.INBOUND,
-                    polling=True
-                )
```
Comment (danielortega-msft) [inline sdk/communication/azure-communication-phonenumbers/test/test_phone_number_administration_client_async.py:313]: This shouldn't have been removed.

**Review by whisper6284 — APPROVED** [review]
